python/flask/security/audit/flask-cors-misconfiguration.yaml

```diff
@@ -0,0 +1,36 @@
+rules:
+  - id: flask-cors-misconfiguration
+    message: >-
+      Setting 'support_credentials=True' together with 'origin="*"' is a CORS
+      misconfiguration that can allow third party origins to read sensitive
+      data. Using this configuration, flask_cors will dynamically reflects the
+      Origin of each request in the Access-Control-Allow-Origin header, allowing
+      all origins and allowing cookies and credentials to be sent along with
+      request. It is recommended to specify allowed origins instead of using "*"
+      when setting 'support_credentials=True'.
+    languages:
+      - python
+    severity: WARNING
+    patterns:
+      - pattern-either:
+          - pattern: |
+              @cross_origin(..., origins="*", supports_credentials=True, ...)
+          - pattern: |
+              CORS(..., supports_credentials=True, origins="*", ...)

```

Comment:
```suggestion
          - pattern: |
              @cross_origin(..., origins="*", ..., supports_credentials=True, ...)
          - pattern: |
              CORS(..., supports_credentials=True, ..., origins="*", ...)

```

Reply:
I'm assuming this annotation has other options that can potentially be set in between these two

Reply:
Hi @0xDC0DE,

thank you for your feedback.
I've tried to keep the rule as simple as possible (by using the minimum amount of ellipsis operators) because I've noticed that semgrep is already able to handle cases where there are other options in between. Please take a look here --> https://semgrep.dev/playground/s/wdE9Y

Reply:
Ooh that's great! Thanks for checking.